pkg/container-collection/container-collection.go

```diff
@@ -345,7 +345,7 @@ func (cc *ContainerCollection) LookupOwnerReferenceByMntns(mntns uint64) *metav1
 	cc.containers.Range(func(key, value interface{}) bool {
 		c := value.(*Container)
 		if mntns == c.Mntns {
-			ownerRef, err = c.GetOwnerReference()
+			ownerRef, err = c.GetOwnerReference(nil)
```

Comment:
I was checking if that's ok to pass nil to `GetOwnerReference`, forcing it to only accept in-cluster kubeconfig.

The function LookupOwnerReferenceByMntns is only used by the advise seccomp gadget and only when used via a Custom Resource (CR), via the `kubectl-gadget advise seccomp` command. So it is ok to use a in-cluster kubeconfig only.

But it is not obvious why it works. Maybe add a short comment? We should deprecate and remove the CRD soon in a future PR.

Reply:
It seems that the call to the api-server by GetOwnerReference is done at two different time:
1. by WithKubernetesEnrichment when IG detects a new container. It is stored in the container struct.
2. by the advise seccomp gadget via LookupOwnerReferenceByMntns

Why does LookupOwnerReferenceByMntns need to call the api-server instead of reading the cache in `container.K8s.ownerReference`?

Reply:
> Why does LookupOwnerReferenceByMntns need to call the api-server instead of reading the cache in `container.K8s.ownerReference`?

Ahh right. Since `advice seccomp` is done only by `kubectl-gadget` we don't need the extra `LookupOwnerReferenceByMntns` call. Thanks for pointing this out